### PR TITLE
main-nuxt: eslint

### PR DESCRIPTION
**Pull request by roiLeo**
linter on main-nuxt

### PR type
- [x] Feature

### DONE
- [x] .hound.yml
```yaml
eslint:
  enabled: true
  config_file: .eslintrc
```

### CODE REVIEW

**Comment by roiLeo** [comment]
Preview is working, but not locally. 
I'm starting to run out of ideas.

**Comment by yangwao** [comment]
> Preview is working, but not locally. I'm starting to run out of ideas.

Should I send you my MacBook over post

**Comment by yangwao** [comment]
What about Hound @roiLeo ? Should we respect it or merge it?

**Comment by yangwao** [comment]
Oh got it, try to purge your whole node installation, back in v4-v6-v8 transition I've had some weird issues like you have.

Other idea.. you can run virtualization/VPS and ssh remote tunnel on that instance 😄 I can pay out that costs lol

File: params/components/findComponent.js [hunk 1]
```diff
@@ -1,120 +1,120 @@
...
-import Vote from './Vote.vue';
-import VoteThreshold from './VoteThreshold.vue';
+import { TypeDefInfo } from '@polkadot/types/types'
+import BN from 'bn.js'
+import { createType, getTypeDef } from '@polkadot/types'
+import registry from './typeRegistry'
+import Account from './Account.vue'
+import Amount from './Amount.vue'
+import Balance from './Balance.vue'
+import Bool from './Bool.vue'
```
[the remark this answers is not in the document]
Comment (vikiival) [reply params/components/findComponent.js:11]: Why is this file here 👁️

**Comment by roiLeo** [comment]
> > Preview is working, but not locally. I'm starting to run out of ideas.
> 
> Should I send you my MacBook over post

Yes! this could be the solution :)

> What about Hound @roiLeo ? Should we respect it or merge it?

Hound is not properly configured, it should respect our rules from eslint file.

**Comment by yangwao** [comment]
> > > Preview is working, but not locally. I'm starting to run out of ideas.
> > 
> > 
> > Should I send you my MacBook over post
> 
> Yes! this could be the solution :)
> 
> > What about Hound @roiLeo ? Should we respect it or merge it?
> 
> Hound is not properly configured, it should respect our rules from eslint file.

Let's configure it. It should be in github workflow iirc?

**Comment by roiLeo** [comment]
> Let's configure it. It should be in github workflow iirc?

Can't configure Hound properly: 
- last eslint support version 7.7.0.
- only support JSON or YAML formats for .eslintrc 
- weird bug with eslint-plugin@typescript-eslint

**Comment by yangwao** [comment]
so let's fix it later and open issue on that, we can merge this one?

**Comment by roiLeo** [comment]
> so let's fix it later and open issue on that, we can merge this one?

If you remove .hound.yml file it will work same as before.
We can merge it as it introduce `eslint`, `yarn lint` & lint workflow on nuxt branch.

**Comment by yangwao** [comment]
The advantage of hound was that it opens comments on PR where previous flow put it as a warning on the bottom?

**Comment by roiLeo** [comment]
> The advantage of hound was that it opens comments on PR where previous flow put it as a warning on the bottom?

I think original Hound is reviewing on ES6 syntax or we are working with ES2020/ES11.
Usually with previous flow, if lint check (or any other check) is failed it shouldn't be merged.

**Comment by vikiival** [comment]
So may let's close this and start again?

**Comment by roiLeo** [comment]
> So may let's close this and start again?

See hound list of issues for more informations on TS and eslint support -> https://github.com/houndci/hound/issues 

We have multiple options:
1. remove hound and use eslint workflow
2. keep hound and change eslint rules to match hound spec (semicolon, ES6 etc...)
3. keep hound and downgrade eslint version to 7.7.0 (not sure about this one)

**Comment by vikiival** [comment]
> 1. remove hound and use eslint workflow
I would go for this. When hound is upgraded happy to use it again.
So far it's just noise for me

**Comment by roiLeo** [comment]
> > 1. remove hound and use eslint workflow
> >    I would go for this. When hound is upgraded happy to use it again.
> >    So far it's just noise for me

Found this [workflow](https://github.com/reviewdog/action-eslint) we can try it out using [Reviewdog](https://github.com/reviewdog)

**Comment by yangwao** [comment]
> > > 1. remove hound and use eslint workflow
> > >    I would go for this. When hound is upgraded happy to use it again.
> > >    So far it's just noise for me
> 
> Found this [workflow](https://github.com/reviewdog/action-eslint) we can try it out using [Reviewdog](https://github.com/reviewdog)

Let's see https://github.com/kodadot/nft-gallery/issues/1314 
Hound was randomly found through gh marketplace afaik so, it was just random bang, let's remove it and try reviewdog if it's configurable!

**Pull request closed without merging**